tools_opera_ms/MaxBin-2.2.4/auxiliary/hmmer-3.1b1/easel/miniapps/esl-alimanip.man
.TH "esl-alimanip" 1  "@RELEASEDATE@" "@PACKAGE@ @RELEASE@" "@PACKAGE@ Manual"

.SH NAME
.TP
esl-alimanip - manipulate a multiple sequence alignment

.SH SYNOPSIS

.B esl-alimanip
.I [options]
.I msafile

.SH DESCRIPTION

.pp
.B esl-alimanip
can manipulate the multiple sequence alignment(s) in 
.I msafile
in various ways. Options exist to remove
specific sequences, reorder sequences, designate reference columns
using Stockholm "#=GC RF" markup, and add annotation that numbers
columns. 

The alignments can be of protein or DNA/RNA sequences. All alignments
in the same 
.I msafile
must be either protein or DNA/RNA. The alphabet will be autodetected
unless one of the options 
.I --amino,
.I --dna,
or 
.I --rna 
are given. 

.SH OPTIONS

.TP
.B -h 
Print brief help;  includes version number and summary of
all options, including expert options.

.TP
.BI -o " <f>"
Save the resulting, modified alignment in Stockholm format to a file
.I <f>.
The default is to write it to standard output.

.TP 
.BI --informat " <s>"
Specify that the input alignment be format 
.I <s>.
At preset, the only valid choices for 
.I <s> 
are: 'stockholm', 'pfam', and 'afa'. 'pfam' is a special
case of Stockholm format in which each sequence is placed on a single
line, instead of being interleaved; 'afa' is aligned FASTA. By default
alignments are assumed to be in Stockholm format (either interleaved
or Pfam).

.TP 
.BI --outformat " <s>"
Specify that the output alignment be format 
.I <s>.
Choices for 
.I <s> 
are: 'stockholm', 'pfam', 'a2m', 'psiblast', 'afa'. 
By default the alignment is output in interleaved Stockholm format.

.TP
.B --devhelp
Print help, as with  
.B "-h",
but also include undocumented developer options. These options are not
listed below, are under development or experimental, and are not
guaranteed to even work correctly. Use developer options at your own
risk. The only resources for understanding what they actually do are
the brief one-line description printed when
.B "--devhelp"
is enabled, and the source code.

.SH EXPERT OPTIONS

.TP 
.BI --lnfract " <x>"
Remove any sequences with length less than 
.I <x>
fraction the length of the median length sequence in the alignment.

.TP 
.BI --lxfract " <x>"
Remove any sequences with length more than 
.I <x>
fraction the length of the median length sequence in the alignment.

.TP 
.BI --lmin " <n>"
Remove any sequences with length less than 
.I <n>
residues.

.TP 
.BI --lmax " <n>"
Remove any sequences with length more than 
.I <n>
residues.

.TP 
.BI --detrunc " <n>"
Remove any sequences that have all gaps in the first 
.I <n>
non-gap #=GC RF columns or the last 
.I <n>
non-gap #=GC RF columns.

.TP 
.BI --xambig " <n>"
Remove any sequences that has more than
.I <n>
ambiguous (degenerate) residues.

.TP 
.BI --seq-r " <f>"
Remove any sequences with names listed in file 
.I <f>.
Sequence names listed in 
.I <f>
can be separated by tabs, new lines, or spaces.
The file must be in Stockholm format for this option to work. 

.TP 
.BI --seq-k " <f>"
Keep only sequences with names listed in file 
.I <f>.
Sequence names listed in 
.I <f>
can be separated by tabs, new lines, or spaces.
By default, the kept sequences will remain in the original order
they appeared in 
.I msafile,
but the order from 
.I <f> 
will be used if the 
.B --k-reorder
option is enabled.
The file must be in Stockholm format for this option to work. 

.TP 
.BI --small
With
.B --seq-k 
or
.B --seq-r,
operate in small memory mode. 
The alignment(s) will not be stored in memory, thus
.B --seq-k 
and
.B --seq-r
will be able to work on very large alignments regardless
of the amount of available RAM.
The alignment file must be in Pfam
format and 
.BI --informat " pfam"
and one of
.B --amino,
.B --dna,
or
.B --rna
must be given as well.

.TP 
.BI --k-reorder
With
.BI --seq-k " <f>",
reorder the kept sequences in the output alignment to the order
from the list file
.I <f>.

.TP 
.BI --seq-ins " <n>"
Keep only sequences that have at least 1 inserted residue after 
nongap RF position 
.I <n>.

.TP 
.BI --seq-ni " <n>"
With 
.B --seq-ins
require at least 
.I <n> 
inserted residues in a sequence for it to be kept.

.TP 
.BI --seq-xi " <n>"
With 
.B --seq-ins
allow at most
.I <n> 
inserted residues in a sequence for it to be kept.

.TP 
.BI --trim " <f>"
File 
.I <f>
is an unaligned FASTA file containing truncated versions of each
sequence in the 
.I msafile. 
Trim the sequences in the alignment to match their truncated versions
in 
.I <f>.
If the alignment output format is Stockholm (the default output
format), all per-column (GC) and per-residue (GR) annotation will be
removed from the alignment when
.B --trim
is used. However, if 
.B --t-keeprf 
is also used, the reference annotation (GC RF) will be kept.

.TP 
.B --t-keeprf
Specify that the 'trimmed' alignment maintain the original
reference (GC RF) annotation. Only works in combination with 
.B --trim.

.TP 
.BI --minpp " <x>"
Replace all residues in the alignments for which the posterior
probability annotation (#=GR PP) is less than 
.I <x>
with gaps. The PP annotation for these residues is also converted to
gaps. 
.I <x>
must be greater than 0.0 and less than or equal to 0.95.

.TP 
.BI --tree " <f>"
Reorder sequences by tree order. 
Perform single linkage clustering on the sequences in the alignment
based on sequence identity given the alignment to define a 'tree' 
of the sequences. The sequences in the alignment are reordered
according to the tree, which groups similar sequences together. The
tree is output in Newick format to 
.I <f>.

.TP 
.BI --reorder " <f>"
Reorder sequences to the order listed in file 
.I <f>.
Each sequence in the alignment must be listed in 
.I <f>.
Use
.B --k-reorder
to reorder only a subset of sequences to a subset alignment file. 
The file must be in Stockholm format for this option to work. 

.TP 
.BI --mask2rf " <f>"
Read in the 'mask' file 
.I <f>
and use it to define new #=GC RF annotation for the 
alignment.
.I <f>
must be a single line, with exactly 
.I <alen> 
or 
.I <rflen>
characters, either the full alignment length or the number of nongap #=GC RF characters, respectively.
Each character must be either a '1'
or a '0'. The new #=GC RF markup will contain an 'x' for each column
that is a '1' in lane mask file, and a '.' for each column that is a '0'. 
If the mask is of length
.I <rflen>
then it is interpreted as applying to only nongap RF characters in the
existing RF annotation, all gap RF characters will remain gaps and
nongap RF characters will be redefined as above.

.TP 
.BI --m-keeprf
With 
.B --mask2rf,
do not overwrite existing nongap RF characters that are included by
the input mask as 'x', leave them as the character they are.

.TP 
.BI --num-all 
Add annotation to the alignment numbering all of the columns in the
alignment. 

.TP 
.BI --num-rf 
Add annotation to the alignment numbering the non-gap (non '.') #=GC
RF columns of the alignment. 

.TP 
.BI --rm-gc " <s>"
Remove certain types of #=GC annotation from the alignment. 
.I "<s>" 
must be either "RF", "SS_cons", "SA_cons", or "PP_cons",
at present.

.TP 
.BI --sindi 
Annotate individual secondary structures for each sequence by imposing
the consensus secondary structure defined by the #=GC SS_cons
annotation. 

.TP 
.BI --post2pp 
Update Infernal's cmalign 0.72-1.0.2 posterior probability "POST"
annotation to "PP" annotation, which is read by other miniapps,
including 
.B esl-alimask
and 
.B esl-alistat.

.TP
.B --amino
Assert that the 
.I msafile 
contains protein sequences. 

.TP 
.B --dna
Assert that the 
.I msafile 
contains DNA sequences. 

.TP 
.B --rna
Assert that the 
.I msafile 
contains RNA sequences. 

.SH AUTHOR

Easel and its documentation are @EASEL_COPYRIGHT@.
@EASEL_LICENSE@.
See COPYING in the source code distribution for more details.
The Easel home page is: @EASEL_URL@


tools_opera_ms/MaxBin-2.2.4/auxiliary/hmmer-3.1b1/easel/miniapps/esl-alimanip.itest.pl
#! /usr/bin/perl

# Integrated test of the esl-alimanip miniapp.
#
# Usage:     ./esl-alimanip.itest.pl <esl-alimanip binary> <tmpfile prefix>
# Example:   ./esl-alimanip.itest.pl ./esl-alimanip        foo
#
# EPN, Tue Feb  2 13:19:44 2010

$eslalimanip= shift;
$tmppfx      = shift;

if (! -x "$eslalimanip") { die "FAIL: didn't find esl-alimanip binary $eslalimanip"; }

# <[miniapps]> esl-seqstat --rna -a foo.stk 
#= seq1                       29 
#= seq2                       31 
#= seq3                       20 

open(ALIFILE, ">$tmppfx.stk") || die "FAIL: couldn't open $tmppfx.stk for writing alifile";
print ALIFILE << "EOF";
# STOCKHOLM 1.0
seq1         aaAAAA..AAAA...Cc.cCCCCCC.C..GGGGGgggg
#=GR seq1 PP 5789**..**88...*9.9****88.7..776543210
seq2         ..AAAAa.AAAAaacCcccCCCCCCcCccGGGGG....
#=GR seq2 PP ..********************************....
seq3         ..AAAA..AAAA...C...CCCCCC.C..GGGG-....
#=GR seq3 PP ..5555..4*44...3...888888.8..8899.....
#=GC SS_cons ...............<...<<<<......>>>>>....
#=GC PP_cons ..789*..8877...8...****99.8..99998....
#=GC RF      ..AAAA..AAAA...C...CCCCCC.c..GGGGG....
//
EOF
close ALIFILE;

open(ALIFILE, ">$tmppfx.stk2") || die "FAIL: couldn't open $tmppfx.stk2 for writing alifile";
print ALIFILE << "EOF";
# STOCKHOLM 1.0
seq1         aaAANA..AAAA...Cc.cCCCCCC.C..GGGGGgggg
#=GR seq1 PP 5789**..**88...*9.9****88.7..776543210
seq2         ..AARAa.AAAAaacCcccCCCCCCcCccGGGGG....
#=GR seq2 PP ..********************************....
seq3         ..AAAA..AAAA...C...CCCCCC.C..GGGG-....
#=GR seq3 PP ..5555..4*44...3...888888.8..8899.....
#=GC SS_cons ...............<...<<<<......>>>>>....
#=GC PP_cons ..789*..8877...8...****99.8..99998....
#=GC RF      ..AAAA..AAAA...C...CCCCCC.c..GGGGG....
//
EOF
close ALIFILE;

open(ALIFILE, ">$tmppfx.post.stk") || die "FAIL: couldn't open $tmppfx.oldp.stk for writing alifile";
print ALIFILE << "EOF";
# STOCKHOLM 1.0
seq1              aaAAAA..AAAA...Cc.cCCCCCC.C..GGGGGgggg
#=GR seq1 POSTX.  5789**..**88...*9.9****88.7..776543210
#=GR seq1 POST.X  3717**..**71...*2.3****34.5..239189922
seq2              ..AAAAa.AAAAaacCcccCCCCCCcCccGGGGG....
#=GR seq2 POSTX.  ..********************************....
#=GR seq2 POST.X  ..********************************....
seq3              ..AAAA..AAAA...C...CCCCCC.C..GGGG-....
#=GR seq3 POSTX.  ..5555..4*44...3...888888.8..8899.....
#=GR seq3 POST.X  ..4567..3*99...1...788882.1..0815.....
#=GC SS_cons      ...............<...<<<<......>>>>>....
#=GC RF           ..AAAA..AAAA...C...CCCCCC.c..GGGGG....
//
EOF
close ALIFILE;

open(ALIFILE, ">$tmppfx.afa") || die "FAIL: couldn't open $tmppfx.afa for writing alifile";
print ALIFILE << "EOF";
>seq1
aaAAAA..AAAA...Cc.cCCCCCC.C..GGGGGgggg
>seq2
..AAAAa.AAAAaacCcccCCCCCCcCccGGGGG....
>seq3
..AAAA..AAAA...C...CCCCCC.C..GGGG-....
EOF
close ALIFILE;

open(LISTFILE, ">$tmppfx.list") || die "FAIL: couldn't open $tmppfx.stk for writing alifile";
print LISTFILE << "EOF";
seq1
seq2
EOF
close LISTFILE;

open(LISTFILE, ">$tmppfx.list2") || die "FAIL: couldn't open $tmppfx.stk for writing alifile";
print LISTFILE << "EOF";
seq3
seq2
seq1
EOF
close LISTFILE;

open(SEQFILE, ">$tmppfx.trim.fa") || die "FAIL: couldn't open $tmppfx.trim.fa for writing alifile";
print SEQFILE << "EOF";
>seq1
ccCCCCCCCGGGGGg
>seq2
CCCCcCccGG
>seq3
AACCCCCCCCG
EOF
close SEQFILE;

open(MASKFILE, ">$tmppfx.rfmask") || die "FAIL: couldn't open $tmppfx.trim.fa for writing alifile";
print MASKFILE << "EOF";
011011110101110110101
EOF
close MASKFILE;

open(MASKFILE, ">$tmppfx.amask") || die "FAIL: couldn't open $tmppfx.trim.fa for writing alifile";
print MASKFILE << "EOF";
01011011011010111011111101010010001100
EOF
close MASKFILE;

$output = `$eslalimanip -h`;
if ($? != 0)                                     { die "FAIL: esl-alimanip failed unexpectedly"; }
if ($output !~ /Usage: esl-alimanip/)            { die "FAIL: help output not right"; }

$output = `$eslalimanip --devhelp`;
if ($? != 0)                                     { die "FAIL: esl-alimanip failed unexpectedly"; }
if ($output !~ /Usage: esl-alimanip/)            { die "FAIL: devhelp output not right"; }


$output = `$eslalimanip --rna $tmppfx.stk 2>&1`;
if ($? != 0)                                                                    { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /\#=GR seq1 PP 5789\*\*..\*\*88...\*9.9\*\*\*\*88.7..776543210/) { die "FAIL: alignment manipulated incorrectly"; }

system("$eslalimanip -o $tmppfx.o.stk --rna $tmppfx.stk > /dev/null");
if ($? != 0)                                                                    { die "FAIL: esl-alimanip failed unexpectedly";}
$output = `cat $tmppfx.o.stk`;
if ($output !~ /\#=GR seq1 PP 5789\*\*..\*\*88...\*9.9\*\*\*\*88.7..776543210/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --dna $tmppfx.stk 2>&1`;
if ($? != 0)                                                                    { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /\#=GR seq1 PP 5789\*\*..\*\*88...\*9.9\*\*\*\*88.7..776543210/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --amino $tmppfx.stk 2>&1`;
if ($? != 0)                                                                    { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /\#=GR seq1 PP 5789\*\*..\*\*88...\*9.9\*\*\*\*88.7..776543210/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --informat stockholm $tmppfx.stk 2>&1`;
if ($? != 0)                                                                    { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /\#=GR seq1 PP 5789\*\*..\*\*88...\*9.9\*\*\*\*88.7..776543210/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --informat pfam $tmppfx.stk 2>&1`;
if ($? != 0)                                                                    { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /\#=GR seq1 PP 5789\*\*..\*\*88...\*9.9\*\*\*\*88.7..776543210/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --informat afa $tmppfx.afa 2>&1`;
if ($? != 0)                                                                    { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /--AAAA--AAAA---C---CCCCCC-C--GGGG-----/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --outformat afa $tmppfx.stk 2>&1`;
if ($? != 0)                                             { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /AAAAAA--AAAA---CC-CCCCCCC-C--GGGGGGGGG/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --outformat psiblast $tmppfx.stk 2>&1`;
if ($? != 0)                                             { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /aaAAAA--AAAA---Cc-cCCCCCC-C--GGGGGgggg/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --outformat a2m $tmppfx.stk 2>&1`;
if ($? != 0)                                    { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /aaAAAAAAAACccCCCCCCCGGGGGgggg/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --lnfract 0.9 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --lnfract 0.9 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq3
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --lxfract 1.02 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq2
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --lmin 30 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq1 and seq3
if ($output =~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --detrunc 1 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq3
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --xambig 0 $tmppfx.stk2 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq1 and seq2
if ($output =~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --seq-k $tmppfx.list $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq3
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --seq-k $tmppfx.list --informat pfam --small $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq3
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --k-reorder --seq-k $tmppfx.list $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq3
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --seq-r $tmppfx.list $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq1 and seq2
if ($output =~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --seq-r $tmppfx.list --informat pfam --small $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should remove seq1 and seq2
if ($output =~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --seq-ins 9 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should keep seq1 and seq 2
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --seq-ins 9 --seq-ni 3 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should keep only seq2
if ($output =~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --seq-ins 9 --seq-xi 2 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should only keep seq1
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output =~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --trim $tmppfx.trim.fa $tmppfx.stk 2>&1`;
if ($? != 0)                                                    { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /seq1\s+----------------C-CCCCCCC-C--GGGGGG---/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq3\s+----------AA---C---CCCCCC-C--G--------/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --minpp 0.9 $tmppfx.stk 2>&1`;
if ($? != 0)                                                     { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /seq1\s+----AA--AA-----C---CCCC---------------/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq3\s+---------A----------------------------/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --reorder $tmppfx.list2 $tmppfx.stk 2>&1`;
if ($? != 0)           { die "FAIL: esl-alimanip failed unexpectedly";}
# should keep all seqs
if ($output !~ /seq1/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq2/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /seq3/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --mask2rf $tmppfx.rfmask $tmppfx.stk 2>&1`;
if ($? != 0)                                                          { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /#=GC RF      ...xx...xxxx.......x.xxx..x..x.x.x..../) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --m-keeprf --mask2rf $tmppfx.rfmask $tmppfx.stk 2>&1`;
if ($? != 0)                                                          { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /#=GC RF      ...AA...AAAA.......C.CCC..c..G.G.G..../) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --mask2rf $tmppfx.amask $tmppfx.stk 2>&1`;
if ($? != 0)                                                          { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /#=GC RF      .x.xx.xx.xx.x.xxx.xxxxxx.x.x..x...xx../) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --m-keeprf --mask2rf $tmppfx.amask $tmppfx.stk 2>&1`;
if ($? != 0)                                                          { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /#=GC RF      .x.AA.xx.AA.x.xCx.xCCCCC.x.x..G...xx../) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --num-all $tmppfx.stk 2>&1`;
if ($? != 0)                                                          { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /#=GC COLX.   00000000011111111112222222222333333333/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /#=GC COL.X   12345678901234567890123456789012345678/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --num-rf $tmppfx.stk 2>&1`;
if ($? != 0)                                                          { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output !~ /#=GC RFCOLX. ..0000..0000...0...111111.1..11122..../) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /#=GC RFCOL.X ..1234..5678...9...012345.6..78901..../) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --rm-gc RF $tmppfx.stk 2>&1`;
if ($? != 0)              { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output =~ /#=GC RF/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --rm-gc SS_cons $tmppfx.stk 2>&1`;
if ($? != 0)                   { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output =~ /#=GC SS_cons/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --rm-gc PP_cons $tmppfx.stk 2>&1`;
if ($? != 0)                   { die "FAIL: esl-alimanip failed unexpectedly";}
if ($output =~ /#=GC PP_cons/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --sindi $tmppfx.stk 2>&1`;
if ($? != 0)                                                          { die "FAIL: esl-alimanip failed unexpectedly";  }
if ($output !~ /#=GR seq1 SS :::::::::::::::<---<<<<______>>>>>::::/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /#=GR seq3 SS :::::::::::::::::::<<<<______>>>>:::::/) { die "FAIL: alignment manipulated incorrectly"; }

$output = `$eslalimanip --rna --post2pp $tmppfx.post.stk 2>&1`;
if ($? != 0)                                                                    { die "FAIL: esl-alimanip failed unexpectedly";  }
if ($output !~ /#=GR seq1 PP 588\*\*\*..\*\*98...\*9.9\*\*\*\*88.8..777554310/) { die "FAIL: alignment manipulated incorrectly"; }
if ($output !~ /#=GR seq3 PP ..5666..4\*55...3...999998.8..899\*...../)         { die "FAIL: alignment manipulated incorrectly"; }

print "ok\n"; 
unlink "$tmppfx.stk";
unlink "$tmppfx.stk2";
unlink "$tmppfx.o.stk";
unlink "$tmppfx.post.stk";
unlink "$tmppfx.afa";
unlink "$tmppfx.list";
unlink "$tmppfx.list2";
unlink "$tmppfx.trim.fa";
unlink "$tmppfx.rfmask";
unlink "$tmppfx.amask";
exit 0;
